### DISCUSSION TITLE
Using Ruleset Function In Custom Function

### DISCUSSION THREAD

**Discussion by OllieTho** [stoplightio/spectral#2323] [Q&A]
I'm running into problems trying to import a function from the oas ruleset into my custom function.

For context, I'm overriding a rule in the included oas ruleset (oas3-unused-component) and want the same behavior as the original rule, but only use the rule in files that contain a 'paths' attribute.  I've made a custom function and added the check for 'paths', but after succeeding that check, I want to call the original function oasUnusedComponents.  I've tried importing it several ways to no success, but this is the closest I've come:

```
import {oasUnusedComponent} from '@stoplight/spectral-rulesets/dist/oas/functions';

module.exports = function(targetVal, _opts, paths, ...args) {
    if (targetVal === void 0 || targetVal.length == 0) {
        return [];
    }

    if ('paths' in targetVal && targetVal['paths']) {
        return oasUnusedComponent(targetVal, _opts, paths);
    }

    return [];
}
```

I've tried modeling this off of the examples for core functions such as 'truthy' but this way does not seem to be working.  Any help would be much appreciated!  Thanks!

**Comment by OllieTho** [comment]
The solution I've found that works is adding this inside the module.exports function:
`const {oasUnusedComponent} = require('@stoplight/spectral-rulesets/dist/oas/functions');`

Since require() is a node.js function, this will only work on the cli version, so the issue will persist if you are reliant on the browser version.

**Reply by P0lip** [reply]
Using an OAS ruleset function in a different ruleset is not something we support and it'll break at some point.
Those functions may potentially change depending on the rules we use them for, therefore you may be exposed to breaking changes we introduce to them. Such changes wouldn't be listed anywhere, because it's just the rules that are semantically versioned, In other words, if we introduced a breaking change to a rule, that would result in a major release of @stoplight/spectral-rulesets, but a breaking change to a function itself would not result in a major release granted the behavior of the rule did not change.

Please refer to https://github.com/stoplightio/spectral/discussions/2398#discussioncomment-4926211 for more details.
